tests/DependencyInjection/NeustaConverterExtensionTest.php

```diff
@@ -123,6 +126,122 @@ public function test_with_mapped_context(): void
         self::assertSame('age', $ageInYearsPopulator->getArgument('$contextProperty'));
     }
 
+    public function test_with_converting_populator(): void
+    {
+        $container = $this->buildContainer([
+            'populators' => [
+                'foobar' => [
+                    'converter' => GenericConverter::class,
+                    'property' => [
+                        'targetTest' => 'sourceTest',
+                    ],
+                ],
+            ],
+        ]);
+
+        // converter
+        $populator = $container->getDefinition('foobar');
+
+        self::assertSame(ConvertingPopulator::class, $populator->getClass());
+        self::assertTrue($populator->isPublic());
+        self::assertInstanceOf(TypedReference::class, $populator->getArgument('$converter'));
+        self::assertSame('targetTest', $populator->getArgument('$targetPropertyName'));
+        self::assertSame('sourceTest', $populator->getArgument('$sourcePropertyName'));
+    }
+
+    public function test_with_array_converting_populator(): void
+    {
+        $container = $this->buildContainer([
+            'populators' => [
+                'foobar' => [
+                    'converter' => GenericConverter::class,
+                    'property' => [
+                        'targetTest' => 'sourceTest',
+                    ],
+                ],
+            ],
+        ]);
+
+        // converter
+        $populator = $container->getDefinition('foobar');
+
+        self::assertSame(ConvertingPopulator::class, $populator->getClass());
+        self::assertTrue($populator->isPublic());
+        self::assertInstanceOf(TypedReference::class, $populator->getArgument('$converter'));
+        self::assertSame('targetTest', $populator->getArgument('$targetPropertyName'));
+        self::assertSame('sourceTest', $populator->getArgument('$sourcePropertyName'));
+    }
+
+    public function test_with_array_converting_populator_with_inner_property(): void
+    {
+        $container = $this->buildContainer([
+            'populators' => [
+                'foobar' => [
+                    'class' => ArrayConvertingPopulator::class,
+                    'converter' => GenericConverter::class,
+                    'property' => [
+                        'targetTest' => 'sourceTest',
+                        'itemProperty' => 'value',
```

Comment:
I don't think this config is good UX. 

The first entry is `target property name` => `source property name` (custom key with custom value) and the second is `config key name` => `config key value` (specific config key with custom value), so it's mixing two different kinds of configuration here.

Reply:
See #56